### PR TITLE
feat: add FlexBox component

### PR DESCRIPTION
**Pull request by chambo-e**
## Summary

This will allow us to remove loads of `<Box display="flex"`, we currently have ~200

This component does not inherits xstyled props, so it should only be used to replace "pure" flex Box for now

We need to have a way to handle spacing (p and m) and we should be able to replace all xstyled usage

## Type

- Feature

### CODE REVIEW

File: src/components/FlexBox/__stories__/index.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+<Meta title="Components/FlexBox" component={FlexBox} />
+
+# FlexBox
+
+<Description>
+  `FlexBox` a flex container, it accepts the same props as the css spec
+</Description>
+
+<Canvas>
+  <Story name="Container">
```
Comment (DorianMaliszewski) [inline src/components/FlexBox/__stories__/index.stories.mdx:19]: Basic is the first story name 😄

**Review by DorianMaliszewski — CHANGES_REQUESTED** [review]

**Comment by DorianMaliszewski** [comment]
Capture d’écran 2021-05-31 à 13 04 08
Is this normal ? And in the page the ArgsTables are empty IDK why ??

**Comment by chambo-e** [comment]
> Capture d’écran 2021-05-31 à 13 04 08
> 
> Is this normal ? And in the page the ArgsTables are empty IDK why ??

@DorianMaliszewski what do you mean ? The `Show Code` button ?

**Comment by vincentaudebert** [comment]
Any idea why we lose a bit of coverage on Toaster? 🤔

**Review by vincentaudebert — APPROVED** [review]

**Comment by DorianMaliszewski** [comment]
> > Capture d’écran 2021-05-31 à 13 04 08
> > Is this normal ? And in the page the ArgsTables are empty IDK why ??
> 
> @DorianMaliszewski what do you mean ? The `Show Code` button ?

Yeah there a blank space below

**Comment by chambo-e** [comment]
> Any idea why we lose a bit of coverage on Toaster? 🤔

@vincentaudebert I think it's a codecov error, if you go on the website they show no change on these files

**Comment by chambo-e** [comment]
> > > Capture d’écran 2021-05-31 à 13 04 08
> > > Is this normal ? And in the page the ArgsTables are empty IDK why ??
> > 
> > 
> > @DorianMaliszewski what do you mean ? The `Show Code` button ?
> 
> Yeah there a blank space below

@DorianMaliszewski no idea, I don't have it 😄 

Screenshot 2021-06-01 at 12 06 18

**Review by matthprost — APPROVED** [review]

**Pull request merged**